### PR TITLE
Dramatiq integration from @jacobsvante

### LINKED ISSUE #3387: feat: sentry-dramatiq integration
### Problem Statement

5 years ago I created [sentry-dramatiq](https://github.com/jacobsvante/sentry-dramatiq) and there seems to be some active users. I am no longer maintaining the library as I no use Python.

### Solution Brainstorm

Would you consider integrating it into sentry-sdk or perhaps taking over the maintenance of the project?

### PR DESCRIPTION
**Pull request by antonpirker**
This is the code of [sentry-dramatiq](https://github.com/jacobsvante/sentry-dramatiq). 

As described in this GitHub issue (https://github.com/getsentry/sentry-python/issues/3387) @jacobsvante,  the original maintainer of this integations, is not doing any Python anymore and wants to donate his integration to Sentry so we can take care of it. 

This PR adds the current version of the `DramatiqIntegration` to our repo. 

Fixes #3387

Documentation PR: https://github.com/getsentry/sentry-docs/pull/11015

### CODE REVIEW

**Comment by antonpirker** [comment]
Tested with this example project:
https://github.com/antonpirker/testing-sentry/tree/main/test-dramatiq

**Comment by antonpirker** [comment]
Errors with stack traces and local vars are there:
Screenshot 2024-08-07 at 12 56 54

**Comment by antonpirker** [comment]
Additional dramatiq information is  in the `contexts` and an `extra`:
Screenshot 2024-08-07 at 12 56 36

File: sentry_sdk/integrations/dramatiq.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            middleware = list(middleware)
+
+        if integration is not None:
+            assert SentryMiddleware not in (
+                m.__class__ for m in middleware
+            ), "Sentry middleware must not be passed in manually to broker"
+            middleware.insert(0, SentryMiddleware())
+
+        kw["middleware"] = middleware
+        # raise Exception([args, kw])
```
Comment (sentrivana) [inline sentry_sdk/integrations/dramatiq.py:75]:
```suggestion
```

File: sentry_sdk/integrations/dramatiq.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+        try:
+            middleware = kw.pop("middleware")
+        except KeyError:
+            # Unfortunately Broker and StubBroker allows middleware to be
+            # passed in as positional arguments, whilst RabbitmqBroker and
+            # RedisBroker does not.
+            if len(args) > 0:
+                assert len(args) < 2
+                middleware = None if args[0] is None else args[0]
```
Comment (sentrivana) [inline sentry_sdk/integrations/dramatiq.py:58]:
```suggestion
                middleware = args[0]
```

File: sentry_sdk/integrations/dramatiq.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        integration = sentry_sdk.get_client().get_integration(DramatiqIntegration)
+
+        try:
+            middleware = kw.pop("middleware")
+        except KeyError:
+            # Unfortunately Broker and StubBroker allows middleware to be
+            # passed in as positional arguments, whilst RabbitmqBroker and
+            # RedisBroker does not.
+            if len(args) > 0:
+                assert len(args) < 2
```
Comment (sentrivana) [inline sentry_sdk/integrations/dramatiq.py:57]: Maybe let's just check `if len(args) == 1` and remove the assert?

File: sentry_sdk/integrations/dramatiq.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        max_request_body_size = client.options["max_request_body_size"]
+        if (
+            max_request_body_size == "never"
+            or (max_request_body_size == "small" and content_length > 10**3)
+            or (max_request_body_size == "medium" and content_length > 10**4)
+        ):
+            data = AnnotatedValue(
+                "",
+                {"rem": [["!config", "x", 0, content_length]], "len": content_length},
+            )
```
Comment (sentrivana) [inline sentry_sdk/integrations/dramatiq.py:176]: Wouldn't this get trimmed in the serializer anyway?

**Review by sentrivana — APPROVED** [review]
See some small comments. Overall LGTM. Thanks for porting <3

**Comment by jacobsvante** [comment]
Thank you so much for integrating it ❤️

**Pull request merged**